src/main/cpp/src/xxhash64.cu

```diff
@@ -296,27 +320,185 @@ class device_row_hasher {
 
   /**
    * @brief Computes the hash value of an element in the given column.
+   *
+   * When the column is non-nested, this is a simple wrapper around the element_hasher.
+   * When the column is nested, this uses a seed value to serially compute each
+   * nested element, with the output hash becoming the seed for the next value.
+   * This requires constructing a new hash functor for each nested element,
+   * using the new seed from the previous element's hash. The hash of a null
+   * element is the input seed (the previous element's hash).
    */
   class element_hasher_adapter {
    public:
-    template <typename T, CUDF_ENABLE_IF(cudf::column_device_view::has_element_accessor<T>())>
+    class element_hasher {
+     private:
+      Nullate const _check_nulls;
+      hash_value_type const _seed;
+
+     public:
+      __device__ element_hasher(Nullate check_nulls, hash_value_type seed)
+        : _check_nulls(check_nulls), _seed(seed)
+      {
+      }
+
+      template <typename T, CUDF_ENABLE_IF(cudf::column_device_view::has_element_accessor<T>())>
```

Comment:
NIT:
These two methods seems being copied from `element_hasher`, and I prefer to leveraging `element_hasher` here instead of implementing them here again.

Reply:
In `cudf::experimental::row::hash::element_hasher`, the seed is of type `uint32_t`, whereas in `xxhash64`, the seed is of type `int64_t`.